### PR TITLE
feat: send and receive rpcs for wallets

### PR DESCRIPTION
**Pull request by jackstar12**

### CODE REVIEW

File: boltzrpc/boltzrpc.proto [hunk 1]
```diff
@@ -132,6 +132,16 @@ service Boltz {
      */
     rpc RemoveWallet (RemoveWalletRequest) returns (RemoveWalletResponse);
 
+    /*
+    Send coins from a wallet.
+     */
+    rpc WalletSend (WalletSendRequest) returns (WalletSendResponse);
+
+    /*
+    Get a new address to receive coins.
```
Comment (michael1011) [inline boltzrpc/boltzrpc.proto:141]:
```suggestion
    Get a new address of the wallet.
```

File: boltzrpc/boltzrpc.proto [hunk 2]
```diff
@@ -566,6 +578,8 @@ message CreateChainSwapRequest {
     // Boltz does not accept 0-conf for Liquid transactions with a fee of 0.01 sat/vByte;
     // when `lockup_zero_conf` is enabled, a fee of 0.1 sat/vByte will be used for Liquid lockup transactions
     optional bool lockup_zero_conf = 9;
+    // fee rate to use when sending from internal wallet
```
Comment (michael1011) [inline boltzrpc/boltzrpc.proto:581]:
```suggestion
    // Fee rate to use when sending from internal wallet
```

File: boltzrpc/boltzrpc.proto [hunk 3]
```diff
@@ -698,6 +712,26 @@ message RemoveWalletRequest {
     uint64 id = 1;
 }
 
+message WalletSendRequest {
+    uint64 id = 1;
+    string address = 2;
+    uint64 amount = 3;
+    // fee rate to use for the transaction
```
Comment (michael1011) [inline boltzrpc/boltzrpc.proto:719]:
```suggestion
    // Fee rate to use for the transaction
```

File: boltzrpc/boltzrpc.proto [hunk 4]
```diff
@@ -698,6 +712,26 @@ message RemoveWalletRequest {
...
+message WalletSendResponse {
+    string tx_id = 1;
+}
+
+message WalletReceiveRequest {
+    uint64 id = 1;
+}
+
+message WalletReceiveResponse {
+    string address = 2;
```
Comment (michael1011) [inline boltzrpc/boltzrpc.proto:732]:
```suggestion
    string address = 1;
```

File: cmd/boltzcli/commands.go [hunk 5]
```diff
@@ -1879,6 +1896,47 @@ func removeWallet(ctx *cli.Context) error {
...
+
+	if satPerVbyte := ctx.Float64("sat-per-vbyte"); satPerVbyte != 0 {
+		request.SatPerVbyte = &satPerVbyte
+	}
+
+	txId, err := client.WalletSend(request)
+	if err != nil {
+		return err
+	}
+	fmt.Println("Transaction ID:", txId)
```
Comment (michael1011) [inline cmd/boltzcli/commands.go:1922]: JSON would be more appropriate to allow for scripting

File: cmd/boltzcli/commands.go [hunk 6]
```diff
@@ -1879,6 +1896,47 @@ func removeWallet(ctx *cli.Context) error {
...
+	client := getClient(ctx)
+	walletId, err := getWalletId(ctx, ctx.Args().First())
+	if err != nil {
+		return err
+	}
+	address, err := client.WalletReceive(*walletId)
+	if err != nil {
+		return err
+	}
+	fmt.Println("Address:", address)
```
Comment (michael1011) [inline cmd/boltzcli/commands.go:1936]: Same here

File: rpcserver/router.go [hunk 7]
```diff
@@ -1970,17 +2015,19 @@ func (server *routedBoltzServer) getPairs(pairId boltz.Pair) (*boltzrpc.Fees, *b
...
+	if fee == nil {
+		feeSatPerVbyte, err := server.onchain.EstimateFee(wallet.GetWalletInfo().Currency, allowLowball)
+		if err != nil {
+			return "", err
+		}
+		fee = &feeSatPerVbyte
 	}
-	logger.Infof("Using fee of %f sat/vbyte", feeSatPerVbyte)
-	return wallet.SendToAddress(address, amount, feeSatPerVbyte)
+	logger.Infof("Using fee of %f sat/vbyte", *fee)
```
Comment (michael1011) [inline rpcserver/router.go:2029]:
```suggestion
	logger.Debugf("Using fee of %f sat/vbyte", *fee)
```

File: cmd/boltzcli/commands.go (hunk 5 above)
Comment (jackstar12) [reply cmd/boltzcli/commands.go:1922]: or just the raw output

**Review by michael1011 — APPROVED** [review]

**Pull request merged**